src/SignalR/clients/ts/signalr/src/HttpConnection.ts

```diff
@@ -497,6 +488,13 @@ export class HttpConnection implements IConnection {
             this.logger.log(LogLevel.Information, "Connection disconnected.");
         }
 
+        if (this.sendQueue) {
+            this.sendQueue.stop().catch((e) => {
+                this.logger.log(LogLevel.Error, `TransportSendQueue.stop() threw error '${e}'.`);
+            });
+            this.sendQueue = undefined;
+        }
+
```

Comment:
While we're cleaning up random things, can we move the `this.connectionStarted = false;` above the `if (this.onclose && this.connectionStarted) {`? I wrote it that way, but don't know why. It shouldn't be dependent on whether the onclose callback was hooked. Thankfully, HubConnection always does hook the onclose callback, but still.

Reply:
I moved the `if (this.onclose)` check instead. We don't want to run the `onclose` function if the connection never started.

Reply:
> We don't want to run the onclose function if the connection never started.

Yeah. That's the whole point of that logic. We don't need to check if connectionStarted is true in order to set it to false. Both ways work, but it does seem a little weird to put the if block inside the try block.